### ISSUE TITLE
Token Revocation Endpoint is not getting called in case of Public Client

### ISSUE THREAD

**Issue by tapasbose** [spring-projects/spring-authorization-server#999]
Hello,
In my configuration, I have set `clientAuthenticationMethod(ClientAuthenticationMethod.NONE)`, also I have 
```
.tokenRevocationEndpoint(
    (OAuth2TokenRevocationEndpointConfigurer tokenRevocationEndpoint) -> tokenRevocationEndpoint
        .revocationResponseHandler((HttpServletRequest request, HttpServletResponse response, Authentication authentication) -> {
            // my logic

            response.setStatus(HttpStatus.OK.value());
        })
)
```
It seems neither `revocationResponseHandler` is called nor `OAuth2TokenRevocationEndpointFilter.doFilterInternal`.

They are called if I use `clientAuthenticationMethod(ClientAuthenticationMethod.CLIENT_SECRET_POST)`.

I was wondering if it is a bug or token revocation cannot be done in case the client is Public.

Regards

**Comment by jgrandja** [comment]
@stefanocke I don't agree. It does not explicitly state that public clients are allowed.

Regardless, allowing public clients on the token revocation endpoint goes against the "secure by default" principle, which is not what this project wants to adhere to.

If you require unauthenticated access for clients to any of the endpoints, **it's the applications responsibility to configure.**
